docker/puppetserver-standalone/docker-entrypoint.sh

```diff
@@ -10,4 +10,13 @@ for f in /docker-entrypoint.d/*.sh; do
     "$f"
 done
 
+if [ -d /docker-custom-entrypoint.d/ ]; then
+    chmod +x /docker-custom-entrypoint.d/*.sh
+    sync
+    for f in /docker-custom-entrypoint.d/*.sh; do
+        echo "Running $f"
+        "$f"
+    done
```

Comment:
Does this work if the directory exists but is empty? should we guard against that?

Reply:
It **does** error when the directory exists but is empty; good catch...I assumed it iterated 0 times.

Do we know of a good way to do that?

Reply:
Looks like when I was playing around with this a while back I did 
```
if [[ -n "$(ls -A /docker-custom-entrypoint.d)" ]]; then
...
fi
```

Reply:
Ok, there's a few ways to do this I see.

For reference here's what postgres does: https://github.com/docker-library/postgres/blob/87b15b6c65ba985ac958e7b35ba787422113066e/9.6/docker-entrypoint.sh#L145-L163

I like generalizing the pattern-match and then adding a `case` statement, because it means we don't need to check if the directory exists beforehand, we don't need subshells, and it's clear/easy to extend support for other file types in the future (and it doesn't require `bash`, which is a nice bonus).

So this:
```
for f in /docker-custom-entrypoint.d/; do
  case "$f" in
    *.sh) echo "Running $f"; "$f" ;;
    *) ;;
  esac
done
```

Reply:
Updated with a `find`-based solution because there's a surprising number of ways this can fail.

I also wanted to avoid doing a `sync` in a loop, but I opted to put the `chmod` in a loop so we can gracefully handle non-`.sh` files in the directory without having to silence errors from `chmod +x /docker-custom-entrypoint.d/*.sh`.